lib/src/main/java/com/cloudbees/groovy/cps/impl/CallEnv.java

```diff
@@ -57,6 +60,11 @@ public CallEnv(Env caller, Continuation returnAddress, SourceLocation loc, int l
         } else {
             types = Maps.newHashMapWithExpectedSize(localsCount);
         }
+        depth = (caller instanceof  DepthTrackingEnv) ? depth = ((DepthTrackingEnv) caller).getDepth() + 1 : 1;
+
+        if (depth > DepthTrackingEnv.MAX_LEGAL_DEPTH) {
+            throw new StackOverflowError("Excessively nested blocks/function calls exceed limit of"+DepthTrackingEnv.MAX_LEGAL_DEPTH+":"+depth);
```

Comment:
need a space after "of". But do you think there is a way in here to point to the user's code?

Reply:
Yes, I have a commit locally that fixes the space issue (pushing it publicly now) and I'm working on how to point the user's code at it